### PR TITLE
chore: Switch tekton context to use GitLab

### PR DESCRIPTION
**Pull request by abayer**
Once this is ready, I'll disable the `github` and `gitlab` contexts, since those will be covered by the `jenkins` and `tekton` contexts respectively.

Signed-off-by: Andrew Bayer <andrew.bayer@gmail.com>

### CODE REVIEW

**Comment by abayer** [comment]
/retest

Flaky test gets fixed in #1109, fyi.

**Comment by abayer** [comment]
The `go-scm` repository creation on GitLab is failing with the issue https://github.com/jenkins-x/go-scm/pull/192 was supposed to fix. Gonna look into that.

File: pkg/scmprovider/pull_requests.go [hunk 1]
```diff
@@ -35,15 +35,15 @@ func (c *Client) populateFields(ctx context.Context, pr *scm.PullRequest, owner,
...
+	// If we have an ID but no name or no clone URI, we've probably got a GitLab PR so go get its base information
+	if pr != nil && pr.Base.Repo.ID != "" && (pr.Base.Repo.Name == "" || pr.Base.Repo.Clone == "") {
 		baseRepo, _, err := c.client.Repositories.Find(ctx, pr.Base.Repo.ID)
 		if err != nil {
 			return nil, errors.Wrapf(err, "getting base repository for PR")
 		}
 		pr.Base.Repo = *baseRepo
 	}
-	if pr != nil && pr.Head.Repo.ID != "" && pr.Head.Repo.Name == "" {
+	if pr != nil && pr.Head.Repo.ID != "" && (pr.Head.Repo.Name == "" || pr.Head.Repo.Clone == "") {
```
Comment (hferentschik) [inline pkg/scmprovider/pull_requests.go:46]: if the above is GitLab what case is this then?

Comment (abayer) [reply pkg/scmprovider/pull_requests.go:46]: The above is for the base repo, this one is for the head repo - they're both gitlab cases. But I'm gonna go fix that up in go-scm now and put this on hold for the moment anyway.

Comment (abayer) [reply pkg/scmprovider/pull_requests.go:46]: fix is at https://github.com/jenkins-x/go-scm/pull/199 - doing a verification here now.

**Review by hferentschik — APPROVED** [review]

**Comment by hferentschik** [comment]
Sorry, I was maybe a bit too fast with my approval on this one. What's about squashing this one?

**Comment by abayer** [comment]
Gonna squash selectively, but yeah.

**Comment by abayer** [comment]
I brain farted and lgtm'd. =)

File: test/e2e/helpers.go [hunk 2]
```diff
@@ -235,7 +235,7 @@ func CreateBaseRepository(botUser, approver string, botClient *scm.Client, gitCl
 
 // AddCollaborator adds the approver user to the repo
 func AddCollaborator(approver string, repo *scm.Repository, botClient *scm.Client, approverClient *scm.Client) error {
-	_, alreadyCollaborator, _, err := botClient.Repositories.AddCollaborator(context.Background(), fmt.Sprintf("%s/%s", repo.Namespace, repo.Name), approver, "admin")
+	_, alreadyCollaborator, _, err := botClient.Repositories.AddCollaborator(context.Background(), fmt.Sprintf("%s/%s", repo.Namespace, repo.Name), approver, scm.WritePermission)
```
Comment (hferentschik) [inline test/e2e/helpers.go:238]: :-)

File: bdd/jenkins/ci.sh [hunk 3]
```diff
@@ -164,7 +164,7 @@ function install_tools() {
 # Creates the GKE test cluster
 ###############################################################################
 function create_cluster() {
-	CLUSTER_NAME=$(echo ${BRANCH_NAME,,}-${BUILD_ID,,}-lh-tekton-e2e)
+	CLUSTER_NAME=$(echo ${BRANCH_NAME,,}-${BUILD_ID,,}-lh-jenkins-e2e)
```
Comment (hferentschik) [inline bdd/jenkins/ci.sh:167]: 💯  good catch. I have seen this issue myself, but could not figure out why I sometimes would see a failure. That's what I get for starting off with an existing script and not carefully doing a find&replace first :-)

**Pull request merged**